modules/ossm-architecture.adoc

```diff
@@ -3,18 +3,49 @@
+At a high level, {ProductName} consists of a data plane and a control plane:
 
-* *Envoy proxy* intercepts all inbound and outbound traffic for all services in the service mesh. Envoy is deployed as a sidecar to the relevant service in the same pod.
+The *data plane* is a set of intelligent proxies that intercept and control all inbound and outbound network communication between microservices in the service mesh.
+The data plane is implemented in such a way that it intercepts all inbound (ingress) and outbound (egress) network traffic. The Istio data plane is composed of Envoy sidecar proxies running in the same network space as each service to control all network communication between services.
 
-The *control plane* manages and configures Istiod to enforce proxies to route traffic.
+* *Envoy proxies* are the only Istio components that interact with data plane traffic. All incoming (ingress) and outgoing (egress) network traffic between services flows through the proxies. The Envoy proxy also collects all metrics related to the services in the mesh. Envoy proxies are deployed as sidecars, in the same pod as services, and as well as gateways.
 
-Istiod provides service discovery, configuration and certificate management. It converts high-level routing rules to Envoy configurations and propagates them to the sidecars at runtime.
+** *Sidecar proxies* manage inbound and outbound communication to the workload instance it is attached to.
 
-Secret Discovery Service (SDS) distributes certificates and keys to sidecars directly from Istiod.
+** *Gateways* are proxies operating as load balancers receiving incoming or outgoing HTTP/TCP connections. Gateway configurations are applied to standalone Envoy proxies that are running at the edge of the mesh, rather than sidecar Envoy proxies running alongside your service workloads. You use a Gateway to manage inbound and outbound traffic for your mesh, letting you specify which traffic you want to enter or leave the mesh.
 
-{ProductName} also uses the *istio-operator* to manage the installation of the control plane. An _Operator_ is a piece of software that enables you to implement and automate common activities in your OpenShift cluster. It acts as a controller, allowing you to set or change the desired state of objects in your cluster.
+*** *Ingress-gateway* - Also known as an ingress controller, the Ingress Gateway is a dedicated Envoy proxy that receives and controls traffic entering the service mesh. An Ingress Gateway allows features such as monitoring and route rules to be applied to traffic entering the cluster.
+
+*** *Egress-gateway* - Also known as an egress controller, the Egress Gateway is a dedicated Envoy proxy that manages traffic leaving the service mesh. An Egress Gateway allows features such as monitoring and route rules to be applied to traffic exiting the mesh.
```

Comment:
Functionally they're the same, but since this is an overview I wanted to mention both.  We've had requests to define all of the bits and pieces that make up service mesh, so I wanted to mention both gateways.

Reply:
Sounds good to me.